libs/vectorise/include/vectorise/uint/uint.hpp

```diff
@@ -1038,16 +1039,28 @@ constexpr typename UInt<S>::WideType &UInt<S>::ElementAt(std::size_t n)
 }
 
 template <uint16_t S>
-constexpr uint64_t UInt<S>::TrimmedSize() const
+constexpr uint64_t UInt<S>::TrimmedWideSize() const
 {
   uint64_t ret = WIDE_ELEMENTS;
-  while ((ret > 1) && (wide_[ret - 1] == 0))
+  while ((ret > 0) && (wide_[ret - 1] == 0))
   {
     --ret;
   }
   return ret;
 }
 
+template <uint16_t S>
+constexpr uint64_t UInt<S>::TrimmedSize() const
+{
+  uint64_t wide_size{TrimmedWideSize()};
+  uint64_t remainder{WIDE_ELEMENT_SIZE / ELEMENT_SIZE};
+  while ((remainder > 0) && (base()[remainder - 1] == 0))
```

Comment:
As far as it returns 0 for {0ull} now, please also change the test in ledger/libs/math/tests/unit/basic_math/big_number_tests.cpp:194

Reply:
Thanks for mentioning this. It is fixed in the [32a57a8](https://github.com/fetchai/ledger/pull/2011/commits/32a57a8b2f3434f03cc764a316e3edbf5acaec3e) commit bellow.

It was result for manual extraction of this code from the original #1993 PR.